gpii/node_modules/processReporter/src/ProcessReporter.js

```diff
@@ -0,0 +1,156 @@
+    invokers: {
+        handleRequest: {
+            func: "{processReporter}.get",
+            args: [
+                "{request}"
+            ]
+        }
+    }
+});
+
+fluid.defaults("gpii.processReporter.solution.get", {
+    gradeNames: ["kettle.request.http"],
+    invokers: {
+        handleRequest: {
+            func: "{processReporter}.getProcess",
+            args: ["{request}", "{processReporter}", "{solutionsRegistryDataSource}"]
+        }
+    }
+});
+
+fluid.defaults("gpii.processReporter.nameResolver", {
+    gradeNames: ["fluid.component"],
+    invokers: {
+        resolveName: {
+            funcName: "fluid.identity"
+        }
+    }
+});
+
+gpii.processReporter.fireResponse = function (request, runningSolutions, platformReporter) {
+    request.events.onSuccess.fire({
+        solutions: runningSolutions,
+        OS: platformReporter.reportPlatform()
+    });
+};
+
+fluid.defaults("gpii.processReporter.live", {
+    gradeNames: ["gpii.processReporter.base"],
+    invokers: {
+        get: {
+            funcName: "gpii.processReporter.live.get",
+            args: ["{arguments}.0", "{processReporter}", "{flowManager}.solutionsRegistryDataSource"]
+        },
+        getProcess: {
+            funcName: "gpii.processReporter.live.getProcess",
+            args: ["{arguments}.0", "{processReporter}", "{flowManager}.solutionsRegistryDataSource"]
+        },
+        handleIsRunning: {
+            funcName: "gpii.processReporter.live.handleIsRunning",
+            args: ["{arguments}.0", "{arguments}.1", "{processReporter}"]
+                   // entry, entry id, processReporter
+        }
+    }
+});
+
+gpii.processReporter.live.handleIsRunning = function (entry, entryId, processReporter) {
+    var runningSolution = null;
+    if (!!entry.isRunning) {
+        runningSolution = { id: entryId, isRunning: true };
+        fluid.each (entry.isRunning, function (aMember) {
+            var resolvedName = processReporter.nameResolver.resolveName(aMember.type, "processReporter");
+            var partial = fluid.invokeGradedFunction(resolvedName, aMember);
+            runningSolution.isRunning = runningSolution.isRunning && partial;
+        });
+    }
+    return runningSolution;
+};
+
+gpii.processReporter.live.get = function (request, processReporter, solutionsRegistryDataSource) {
+    var runningSolutions = [];
+    solutionsRegistryDataSource.get({os: processReporter.platformReporter.reportPlatform().id}, function onSuccess(entries) {
+        fluid.each(entries, function (entry, entryId) {
+            var foundEntry = gpii.processReporter.live.handleIsRunning(entry, entryId, processReporter);
+            if  (foundEntry !== null) {
+                runningSolutions.push(foundEntry);
+            }
+        });
+        processReporter.fireResponse(request, runningSolutions);
+    });
+};
+
+gpii.processReporter.live.getProcess = function (request, processReporter, solutionsRegistryDataSource) {
```

Comment:
I don't know if that can be done in a straightforward way without entangling this branch with the huge set of changes of wide scope that are in the GPII-442 branch, that will make review very difficult. If there are definitions that are useful in that branch, it may be easier to copy them in by hand. But perhaps there is some way to do a "cherry-picking" kind of merge that is just scoped to a few files, that won't make the ultimate merge with GPII-442 a trainwreck - I've only really stayed in the shallows of git, so I'm not clear how.